src/json2daisy/json2daisy.py

```diff
@@ -3,282 +3,360 @@
-  replacements['hidupdaterates'] = filter_map_template(components, 'updaterate', key_exclude='default', match_exclude=True)
-
-  license_string = pkg_resources.resource_string(__package__, 'resources/LICENSE').decode('utf-8')
-  replacements['license'] = '/*\n * ' + '\n * '.join([line for line in license_string.split('\n')]) + '\n */'
-
-  component_declarations = list(filter(lambda x: not x.get('default', False), components))
-  component_declarations = list(filter(lambda x: x.get('typename', '') != '', component_declarations))
-  if len(component_declarations) > 0:
-    replacements['comps'] = ";\n  ".join(map(lambda x: x['typename'].format_map(x) + ' ' + x['name'], component_declarations)) + ';'
-  non_class_declarations = list(filter(lambda x: 'non_class_decl' in x, component_declarations))
-  if len(non_class_declarations) > 0:
-    replacements['non_class_declarations'] = "\n".join(map(lambda x: x['non_class_decl'].format_map(x), non_class_declarations))
-  
-  env_opts = {"trim_blocks": True, "lstrip_blocks": True}
-
-  # Ideally, this would be what we use, but we'll need to get the jinja PackageLoader class working
-  # loader = jinja2.PackageLoader(__name__)
-  # env = jinja2.Environment(loader=loader, **env_opts)
-  # rendered_header = env.get_template('daisy.h').render(replacements)
-
-  # This following works, but is really annoying
-  header_str = pkg_resources.resource_string(__name__, os.path.join('templates', 'daisy.h'))
-  header_env = jinja2.Environment(loader=jinja2.BaseLoader(), **env_opts).from_string(header_str.decode('utf-8'))
-
-  rendered_header = header_env.render(replacements)
-  
-  # removing all unnecessary fields
-  for comp in components:
-    if 'map_init' in comp:
-      del comp['map_init']
-    if 'typename' in comp:
-      del comp['typename']
-
-  audio_info = target.get('audio', None)
-  audio_channels = audio_info.get('channels', 2) if audio_info is not None else 2
-
-  # This dictionary contains the necessary information to automatically (or manually) 
-  # write code to interface with the generated board
-  board_info = {
-    'name': target['name'],
-    'components': components,
-    'aliases': target['aliases'],
-    'channels': audio_channels
-  }
-
-  return rendered_header, board_info
+
+    target = board_description_dict
+
+    # flesh out target components:
+    components = target.get('components', {})
+    parents = target.get('parents', {})
+
+    for key in parents:
+        parents[key]['is_parent'] = True
+    components.update(parents)
+
+    seed_defs = os.path.join("resources", 'component_defs.json')
+    patchsm_defs = os.path.join("resources", 'component_defs_patchsm.json')
+    petalsm_defs = os.path.join("resources", 'component_defs_petalsm.json')
+    definitions = {'seed': seed_defs, 'patch_sm': patchsm_defs, 'petal_125b_sm': petalsm_defs}
+    som = target.get('som', 'seed')
+
+    global json_defs_file
```

Comment:
btw I'm not a huge fan (rather the opposite) of global variables in python, but I didn't really see an easy way to refactor this out either (and the goal was style/type checking anyway), so kept it like this for now.